src/main.cpp

```diff
@@ -957,6 +957,12 @@ int main(int argc, char* argv[])
   sigaction(SIGINT, &act, NULL);
   sigaction(SIGTERM, &act, NULL);
 
+  // Signal handler that prints all maps when SIGUSR1 was received.
+  static auto sigusr1_handler = [&]() { bpftrace.print_maps(); };
```

Comment:
How about setting a flag in the signal handler and running `bpftrace.print_maps()` inside main loop? I highly doubt  `print_maps()` is signal safe

Reply:
Ah, I didn't realize we're in racy terrain when using signals. I've updated the handler to set a flag (similar to the current exit signal handler) and print the maps in the main loop.